tests/cpp/test_host_ir_integration.cpp

```diff
@@ -21,55 +23,145 @@ namespace nvfuser {
 namespace hir {
 
 using testing::Contains;
+
 using HostIrEvaluatorTest = NVFuserTest;
 
 // This test manually creates a HostIrContainer with LaunchKernels and runs it
 // using HostIrEvaluator.
 TEST_F(HostIrEvaluatorTest, LaunchKernel) {
   Fusion fusion;
-  FusionGuard fg(&fusion);
-  TensorView* in = makeSymbolicTensor(2);
-  fusion.addInput(in);
-
-  TensorView* out = set(in);
-  fusion.addOutput(out);
+  {
+    FusionGuard fg(&fusion);
+    TensorView* in = makeSymbolicTensor(2);
+    TensorView* out = set(in);
+    fusion.addInput(in);
+    fusion.addOutput(out);
+  }
 
   auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
-  at::Tensor t0 = at::randn({32, 32}, options);
-  auto ke = std::make_unique<KernelExecutor>();
-  ke->setGroupId(0);
-  ke->compile(&fusion, {t0});
+  at::Tensor in_tensor = at::randn({32, 32}, options);
 
   auto hic = std::make_unique<HostIrContainer>();
-  FusionGuard::setCurFusion(hic.get());
+  {
```

Comment:
This change goes to https://github.com/NVIDIA/Fuser/pull/5095, again pending your review.